src/PolyType.Roslyn/Model/EnumerableDataModel.cs

```diff
@@ -41,6 +42,12 @@ public sealed class EnumerableDataModel : TypeDataModel
     /// </summary>
     public required IMethodSymbol? FactoryMethod { get; init; }
 
+    /// <summary>
+    /// Constructor or static factory method whose shape is governed by <see cref="ConstructionStrategy"/>
+    /// and declares either an <see cref="IEqualityComparer{T}"/> or <see cref="IComparer{T}"/> parameter.
+    /// </summary>
+    public required IMethodSymbol? FactoryMethodWithComparer { get; init; }
```

Comment:
Does this need to be a separate field? It seems one could just look at the arguments of `FactoryMethod` to determine if it accepts a comparer.

Reply:
This allows for a runtime decision to call one factory method versus another based on whether or not the user provided a comparer. This supports the use case that I believe exists that some overloads that accept a comparer reject a null argument.